.github/workflows/ci.yml

```diff
@@ -0,0 +1,51 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+  release:
+    types: [published]
+
+jobs:
+  ci:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+        python-version:
+          - name: 3.6
+            toxenv: py36
+          - name: 3.7
+            toxenv: py37
+          - name: 3.8
+            toxenv: py38
+          - name: 3.8
+            toxenv: mypy
+          - name: 3.9
+            toxenv: py39
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version.name }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version.name }}
+    - name: Ensure we have new enough versions to respect python_version
+      run: python -m pip install -U pip
+    - name: Install dependencies
+      run: brew install gmp coreutils
+      if: runner.os == 'macOS'
+    - name: Install elan
+      run: |
+          curl https://raw.githubusercontent.com/Kha/elan/master/elan-init.sh -sSf | sh -s -- --default-toolchain leanprover-community/lean:3.5.1 -y 
```

Comment:
One more random thing I noticed while doing this -- you're running on lean:3.5.1, obviously if you want we could matrix build across say 2 lean versions, the oldest one this supports and then the newest release.